Content.Server/Silicons/Borgs/BorgSystem.Modules.cs

```diff
@@ -214,9 +230,20 @@ private void RemoveProvidedItems(EntityUid chassis, EntityUid uid, BorgChassisCo
         if (!TryComp<HandsComponent>(chassis, out var hands))
             return;
 
+        if (LifeStage(uid) >= EntityLifeStage.Terminating)
+        {
+            foreach (var (hand, item) in component.ProvidedItems)
+            {
+                QueueDel(item);
+                _hands.RemoveHand(chassis, hand, hands);
+            }
+            component.ProvidedItems.Clear();
+            return;
+        }
```

Comment:
This was the actual bug, the rest is mostly adding more asserts and slightly tweaking module selection logic